### PR TITLE
sanitycheck: Allow skipping the 'unrecognized section' test

### PR DESCRIPTION
**Pull request by SebastianBoe**
Allow the user to disable the "unrecognized section" test. I can see
multiple use-cases for disabling the test.

If orphan sections exist and are dynamically or unpredictably named
the unrecognized section test will fail.

If out-of-tree sections exist, one might want to temporarily disable
the "unrecognized section" test until one has made it recognized.

The test is disabled through a CLI flag.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

**Comment by SebastianBoe** [comment]
Our use-case is out-of-tree dynamically created orphan sections.

Any approach that supports this use-case is OK.

File: scripts/sanitycheck [hunk 1]
```diff
@@ -2528,6 +2528,10 @@ def parse_arguments():
         "-S", "--enable-slow", action="store_true",
         help="Execute time-consuming test cases that have been marked "
         "as 'slow' in testcase.yaml. Normally these are only built.")
+    parser.add_argument(
+        "--disable-unrecognized-section-test", action="store_false",
+        default=True,
```
Comment (carlescufi) [inline scripts/sanitycheck]: Shouldn't this default to `False` to keep backwards compatibility with the current behavior?

**Comment by carlescufi** [comment]
CC @myrdyr

File: scripts/sanitycheck (hunk 1 above)
Comment (SebastianBoe) [reply scripts/sanitycheck]: Good catch. Something strange happened, fixing ...

**Review by nashif — APPROVED** [review]

**Pull request merged**